examples/hotrod/services/driver/redis.go

```diff
@@ -35,27 +35,34 @@ import (
 
 // Redis is a simulator of remote Redis cache
 type Redis struct {
-	tracer opentracing.Tracer // simulate redis as a separate process
+	tracer trace.TracerProvider // simulate redis as a separate process
 	logger log.Factory
 	errorSimulator
 }
 
 func newRedis(otelExporter string, metricsFactory metrics.Factory, logger log.Factory) *Redis {
 	return &Redis{
-		tracer: tracing.Init("redis", otelExporter, metricsFactory, logger),
+		tracer: tracing.InitOTEL("redis", otelExporter, metricsFactory, logger),
 		logger: logger,
 	}
 }
 
 // FindDriverIDs finds IDs of drivers who are near the location.
 func (r *Redis) FindDriverIDs(ctx context.Context, location string) []string {
-	if span := opentracing.SpanFromContext(ctx); span != nil {
+	tracer := r.tracer.Tracer("")
```

Comment:
This should be only done once in the constructor.
```suggestion
	tracer := r.tracer.Tracer("redis-manual")
```

Reply:
`func InitOTEL(...)` should do this job. Right?

Reply:
No. InitOTEL returns TracerProvider. Different layers in a service can use provider to obtain differently named Tracers, e.g. if your server endpoint is HTTP using Gorilla framework, the instrumentation may ask for a tracer called "gorilla", whereas a gRPC instrumentation may ask for tracer called "gRPC".